src/view/ViewProvider.ts

```diff
@@ -0,0 +1,382 @@
+import { Uri, Webview, WebviewView, WebviewViewProvider } from "vscode";
+import { getExtContext } from "../config";
+import { Widget, WidgetProperties } from "./Widget";
+import { Option } from "../utils/monads";
+import { Package, Resources } from "./Package";
+
+export interface Message {
+	cmd: string;
+	args: any;
+}
+
+export type Handler<T> = (provider: ViewProvider, args: T) => any;
+
+export class ViewProvider extends Widget implements WebviewViewProvider {
+
+    public static readonly RESOURCES = Resources.fromResources({
```

Comment:
Tried to make this work with files rather than strings but VSCode creates a pretty aggressive sandbox.